hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -27,66 +27,55 @@ See that document for integration overview within the broader top level system.
+Regardless of the flash size underneath, the flash controller maintains the same data resolution as the bus and processor (default 4B).
+The flash physical controller (see section below) is then responsible for bridging that gap.
 
-*  Support controller initiated read, program and erase of flash.
+The protocol controller currently supports the following features:
+
+*  Controller initiated read, program and erase of flash.
    *  Erase can be either of a page, or an entire bank.
-*  Parameterized support for number of flash banks (default to 2)
-*  For each flash bank, parameterized support for number of flash pages (default to 256)
-*  For each flash page, parameterized support for number of words and word size (default to 256 words of 4-bytes each)
-*  Parameterized support for burst writes, up to 64B
-   *  Controller currently does not support page boundary checks; it is thus legal for software to burst write across a page boundary
+*  Support for differentiation between informational and data flash partitions.
+*  Parameterized support for burst program, up to 64B
+   *  Longer programs are supported, however the protocol controller will directly back-pressure the bus.
+*  Flash memory protection at page boundaries
 *  Features to be added if required
-   *  Parameterizable data width
    *  Program verification
-      *  may not be required if flash physical controller supports alternative mechanisms of verification.
+      *  may not be required if flash memory supports alternative mechanisms of verification.
    *  Erase verification
-      *  may not be required if flash physical controller supports alternative mechanisms of verification.
-   *  Parity / ECC support on a per flash page granularity
-      *  may not be required depending on flash reliability or overall system security strategy.
+      *  may not be required if flash memory supports alternative mechanisms of verification.
    *  Flash redundant pages
       *  Flash may contain additional pages used to remap broken pages for yield recovery.
-      *  The storage, loading and security of redundant pages may also be implemented in the physical controller.
-   *  Flash information pages
-      *  Flash may contain additional pages outside of the data banks to hold manufacturing information (such as wafer location).
-      *  Extra logic may not be required if flash information pages is treated as just a separate address.
+      *  The storage, loading and security of redundant pages may also be implemented in the physical controller or flash memory.
+
+Features to be implemented
+
+*  Enhanced memory protection
+*  Life cycle feature support
+*  Key manager feature support
 
 ### Flash Physical Controller Features
 
-As the flash physical controller is highly dependent on flash memory selected, the default flash physical controller simply emulates real flash behavior with on-chip memories.
```

Comment:
removed all of this because this is really a description of `prim_generic_flash` more than anything else and probably should be its own doc.